### DISCUSSION TITLE
[Camera Support]:  Problem receiving  RSTP stream from a Raspberry Pi Cam

### DISCUSSION THREAD

**Discussion by ronniekw** [blakeblackshear/frigate#10305] [General Support] [answered]
### Describe the problem you are having

I'm trying to stream a video from a Raspberry Pi using the following command:
raspberrypi:~ $ libcamera-vid -t 0 --framerate 5 --inline -o - | cvlc stream:///dev/stdin --sout '#rtp{sdp=rtsp://:8554/stream1}' :demux=h264

I'm seeing the video i VLC using: rtsp://@192.168.14.165:8554/stream1

But it does not work with Frigate. 

Using tcpdump i see the following problem during the RTSP-setup:

Working:
VLC -> Rasperberry Cam:  Transport: RTP/AVP;unicast;client_port=56454-56455
Rasperberry Cam ->VLC:  Transport: RTP/AVP/UDP;unicast;client_port=56454-56455;server_port=43665-43666;ssrc=D1AB2858;mode=play

Not working:
Frigate -> Rasperberry Cam:  Transport: RTP/AVP/TCP;unicast;interleaved=0-1
Rasperberry Cam -> Frigate:  Response: RTSP/1.0 461 Unsupported transport\r\n

Config.yml
  PiCamv2: #Name for your comment
    ffmpeg:
      inputs:
        - path: rtsp://192.168.14.165:8554/stream1
          roles:
            - detect
        - path: rtsp://192.168.14.165:8554/stream1
          roles:
            - record
    record: # <----- Enable recording
      enabled: True
      retain:
        days: 7
        mode: all
    detect:
      enabled: true # Detection is disabled
      width: 640 # The Cameras resolution
      height: 480 # The Cameras resolution
      fps: 5

Any sugestions how to get it working?

### Version

0.13.2-6476F8A

### Frigate config file

```yaml
n/a
```

### Relevant log output

```shell
n/a
```

### FFprobe output from your camera

```shell
n/a
```

### Operating system

Debian

### Install method

Docker Compose

### Coral version

USB

### Network connection

Wired

### Camera make and model

Raspberry Cam v2

**Comment by NickM-27** [comment, marked as answer]
your cam is UDP only so you need to use the UDP input arg

**Reply by ronniekw** [reply]
Thanks!

Changing to UDP transport solved the problem.
 
 PiCamv2: #Name for your comment
    ffmpeg:
      input_args: -rtsp_transport udp

**Reply by NickM-27** [reply]
I'd suggest using preset-rtsp-udp
